lldb/source/Plugins/LanguageRuntime/Swift/LLDBMemoryReader.h

```diff
@@ -84,6 +88,10 @@ llvm::Optional<Address> resolveRemoteAddressFromSymbolObjectFile(uint64_t addres
   llvm::Optional<uint64_t> m_local_buffer;
   uint64_t m_local_buffer_size = 0;
 
+  llvm::Optional<std::function<swift::remote::RemoteAbsolutePointer(
```

Comment:
Would it be a layering violation if we just passed the pointer to the reflection context instead of the lambda? Otherwise, does it have to be an Optional?

Reply:
> Would it be a layering violation if we just passed the pointer to the reflection context instead of the lambda? 

You mean call the function directly? `MemoryReader` doesn't know about `ReflectionContext`, but `ReflectionContext` know about `MemoryReader`. The reason I'm using a lambda is to keep `MemoryReader` unaware of  `ReflectionContext`.

> Otherwise, does it have to be an Optional?

It actually doesn't, not sure what my reasoning was for it to be optional.

Reply:
I guess that LLDBMemoryReader could know about it, but I don't mind the function pointer either.